portfolio/src/main/webapp/maps.js

```diff
@@ -0,0 +1,62 @@
+/** Creates a map and adds it to the page. Adds markers to maps with COVID 
+* test sites for CT. 
+*/
+function createMap() {
+  const map = new google.maps.Map(document.getElementById('map'), {
+    center: {
+      lat: 41.05, 
+      lng: -73.53
+    }, 
+    zoom: 8
+  });
+  var requestOptions = {
+    method: 'GET',
+    redirect: 'follow'
+  };
+
+  let state = "connecticut";
+  // gets user's current location to center map and decide what state to load
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      let url = "https://maps.googleapis.com/maps/api/geocode/json?latlng=" + position.coords.latitude + ",";
+      url += position.coords.longitude + "&key=AIzaSyCsIDmQAVTMjvXr6NZBm1O8b9Ve9EWREwk";
+
+      // gets state from coordiates
+      fetch(url).then(response => response.json()).then(result => {
+        state = result.results[7].address_components[0].long_name;
+        state = state.toLowerCase();
+      }); 
+
+      initialLocation = new google.maps.LatLng(position.coords.latitude, position.coords.longitude);
+      map.setCenter(initialLocation);
+    });
+  }
+
+  // call API to return testing site locations for variable state 
+  fetch("https://covid-19-testing.github.io/locations/" + state + "/complete.json", requestOptions)
```

Comment:
This looks like it would need to be updated in the above callback (line 20) when the state is changed.